contract/tools/agd-lib.js

```diff
@@ -0,0 +1,162 @@
+// @ts-check
+import assert from 'node:assert';
+
+const { freeze } = Object;
+
+const agdBinary = 'agd';
```

Comment:
This is a lot of code I see repeated. I think we should have the `agoric` package provide a JS interface to agd, but this suffices for now

Reply:
I looked into moving a bunch of this stuff into its own package but it was a little bumpy so I bailed, but I still hope to do it later.